### PR TITLE
Fix config vars

### PR DESCRIPTION
**Pull request by ret2libc**
This PR essentially reverts https://github.com/radareorg/radare2/pull/15681/files but it introduces a SETBPREF which has a default callback that can "boolify" r2 vars when necessary.

### CODE REVIEW

**Comment by radare** [comment]
ok to merge now?

**Comment by ret2libc** [comment]
Well, if @kazarmy is ok with it, yes.

File: libr/config/config.c [hunk 1]
```diff
@@ -337,22 +337,20 @@ R_API ut64 r_config_get_i(RConfig *cfg, const char *name) {
...
-		if (f & CN_OFFT || f & CN_INT) {
-			if (!strncmp (node->value, "0x", 2)) {
-				return "addr";
-			}
-			return "int";
+	r_return_val_if_fail (node, "");
+
+	if (r_config_node_is_bool (node)) {
+		return "bool";
+	} else if (r_config_node_is_str (node)) {
```
Comment (radare) [inline libr/config/config.c]: no need to else fter return

File: libr/config/config.c [hunk 2]
```diff
@@ -337,22 +337,20 @@ R_API ut64 r_config_get_i(RConfig *cfg, const char *name) {
...
+		return "str";
+	} else if (r_config_node_is_int (node)) {
+		if (!strncmp (node->value, "0x", 2)) {
+			return "addr";
 		}
+		return "int";
+	} else {
```
Comment (radare) [inline libr/config/config.c]: same here

**Comment by ret2libc** [comment]
@radare changes done. It can be merged IMHO.

**Comment by kazarmy** [comment]
> Well, if @kazarmy is ok with it, yes.

No need to wait for me since @radare is closer to the users.

File: libr/include/r_config.h [hunk 3]
```diff
@@ -98,6 +100,19 @@ R_API bool r_config_set_getter (RConfig *cfg, const char *key, RConfigCallback c
 
 R_API void r_config_serialize(R_NONNULL RConfig *config, R_NONNULL Sdb *db);
 R_API bool r_config_unserialize(R_NONNULL RConfig *config, R_NONNULL Sdb *db, R_NULLABLE char **err);
+
+static inline bool r_config_node_is_bool(RConfigNode *node) {
+	return node->flags & CN_BOOL;
+}
+static inline bool r_config_node_is_int(RConfigNode *node) {
```
Comment (kazarmy) [inline libr/include/r_config.h]: Add a blank line between functions?

**Pull request merged**